rcgen/src/lib.rs

```diff
@@ -470,13 +453,32 @@ impl KeyUsagePurpose {
 		]
 		.iter()
 		.filter_map(|key_usage| {
-			let present = key_usage.to_u16() & value != 0;
+			let present = u16::from(*key_usage) & value != 0;
 			present.then_some(*key_usage)
 		})
 		.collect()
 	}
 }
 
+/// Encode a key usage as the value of a BIT STRING as defined by RFC 5280.
+/// [`u16`] is sufficient to encode the largest possible key usage value (two bytes).
+impl From<KeyUsagePurpose> for u16 {
```

Comment:
I dissuaded the OP from this because I thought we'd prefer to keep this interface private.